tests/cpu_bugs/xen_mitigations.pm

```diff
@@ -0,0 +1,349 @@
+            }
+        },
+        "ibrs=on" => {
+            "default" => {
+                expection   => {'xl dmesg' => ['^(XEN) *Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ SSBD-, Other:']},
+                unexpection => {'xl dmesg' => ['']}
+            }
+        },
+        "ibpb=off" => {
+            "default" => {
+                expection   => {'xl dmesg' => ['^(XEN) *Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ SSBD-, Other:$']},
+                unexpection => {'xl dmesg' => ['^(XEN) *Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ SSBD-, Other: IBPB']}
+            }
+        },
+        "ibpb=on" => {
+            "default" => {
+                expection   => {'xl dmesg' => ['^(XEN) *Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ SSBD-, Other: IBPB']},
+                unexpection => {'xl dmesg' => ['^(XEN) *Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ SSBD-, Other:$']}
+            }
+        },
+        "ssbd=off" => {
+            "default" => {
+                expection   => {'xl dmesg' => ['^(XEN) *Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ SSBD-, Other:']},
+                unexpection => {'xl dmesg' => ['^(XEN) *Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ SSBD+, Other:']}
+            }
+        },
+        "ssbd=on" => {
+            "default" => {
+                expection   => {'xl dmesg' => ['^(XEN) *Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ SSBD+, Other:']},
+                unexpection => {'xl dmesg' => ['^(XEN) *Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ SSBD-, Other:']}
+            }
+        },
+        "eager-fpu=off" => {
+            "default" => {
+                expection   => {'xl dmesg' => ['Support for HVM VMs: MSR_SPEC_CTRL RSB MD_CLEAR']},
+                unexpection => {'xl dmesg' => ['Support for HVM VMs: MSR_SPEC_CTRL RSB EAGER_FPU MD_CLEAR']}
+            }
+        },
+        "eager-fpu=on" => {
+            "default" => {
+                expection   => {'xl dmesg' => ['Support for HVM VMs: MSR_SPEC_CTRL RSB EAGER_FPU MD_CLEAR']},
+                unexpection => {'xl dmesg' => ['Support for HVM VMs: MSR_SPEC_CTRL RSB MD_CLEAR']}
+            }
+        },
+        "l1d-flush=off" => {
+            "default" => {
+                expection   => {''},
+                unexpection => {'xl dmesg' => ['Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ SSBD-, Other: .*L1D_FLUSH']},
+            }
+        },
+        "l1d-flush=on" => {
+            "default" => {
+                expection   => {'xl dmesg' => ['Xen settings: BTI-Thunk JMP, SPEC_CTRL: IBRS+ SSBD-, Other: .*L1D_FLUSH']},
+                unexpection => {''},
+            }
+        },
+    },
+  };
+sub do_check {
+    my $secnario = shift;
+    my $foo      = $secnario->{'default'};
+    if ($foo->{'expection'}) {
+        while (my ($cmd, $lines) = each %{$foo->{'expection'}}) {
+            foreach my $l (@{$lines}) {
+                if ($l ne "") {
+                    assert_script_run("$cmd | grep \"$l\"");
```

Comment:
I suggest that the temporary variable $l can be renamed similar to $tmp to avoid singularity.